lib/Doctrine/DBAL/Tools/Console/Command/PingCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Doctrine\DBAL\Tools\Console\Command;
+
+use Doctrine\DBAL\Connection;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+use function is_numeric;
+use function sleep;
+use function sprintf;
+
+class PingCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $defaultName = 'dbal:ping';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Check db is available')
+            ->addOption('limit', null, InputOption::VALUE_REQUIRED, 'Max number of pings to try', '1')
+            ->addOption('sleep', null, InputOption::VALUE_REQUIRED, 'Length of time (seconds) to sleep between pings', '1');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output) : int
+    {
+        $limit = $input->getOption('limit');
+        if (! is_numeric($limit)) {
+            throw new RuntimeException("Option 'limit' must contain an integer value");
+        }
+        $sleep = $input->getOption('sleep');
+        if (! is_numeric($sleep)) {
+            throw new RuntimeException("Option 'sleep' must contain an integer value");
+        }
+
+        return $this->waitForPing($this->getHelper('db')->getConnection(), (int) $limit, (int) $sleep, $output);
+    }
+
+    /**
+     * @return int > 0 for error
+     */
+    private function waitForPing(Connection $conn, int $limit, int $sleep, OutputInterface $output) : int
+    {
+        while (true) {
+            $last = $this->ping($conn, $output);
+            if ($last === 0 || --$limit === 0) {
+                return $last;
+            }
+            sleep($sleep);
+        }
+
+        return $last;
+    }
+
+    /**
+     * @return int > 0 for error
+     */
+    private function ping(Connection $conn, OutputInterface $output) : int
+    {
+        try {
+            if ($conn->ping()) {
+                return 0;
+            }
+
+            $output->writeln('Ping failed');
+
+            return 1;
+        } catch (Throwable $e) {
+            $output->writeln(sprintf('Connection error: <error>%s</error>', $e->getMessage()));
+
+            return 2;
```

Comment:
What difference does it make for the caller? The ping either succeeded or didn't. Note that in DBAL v3, `ping` will throw an exception on failure, so it's not clear how this would have to be reimplemented w/o breaking the existing behavior.

Reply:
It probably makes little difference to the caller, but as there are two different failures it doesnt do any hard to be more informative - note that this becomes the exit code from the process

Reply:
Ok, I've changed the text to just be 'ping failed', so it won't be misleading when DBAL 3 throws on different failure types - anyway the caught exception will say its a connection error or something else.

Reply:
I still don't understand the meaning of the two different return codes and how they will be supported in `3.0`.